src/cmd/compile/internal/syntax/parser.go

```diff
@@ -2018,6 +2018,48 @@ func (p *parser) commClause() *CommClause {
 	return c
 }
 
+// ParseStmtFromScratch is used to get syntax tree node for arbitrary code
+// string. Used while injecting transaction initialization statements
+func ParseStmtFromScratch(s string) Stmt {
+	var pTmp parser
+	src := "package p; func _() { " + s + "; }"
+	// The filename for the inserted statement is specified as tx.go
+	f, err := Parse(NewFileBase("tx.go"), strings.NewReader(src), nil, nil, 0)
+	if err != nil {
+		pTmp.syntaxError("parse error in injecting statement")
+	}
+	return f.DeclList[0].(*FuncDecl).Body.List[0]
+}
+
+// This function adds two statements to the application code:
+// 1. var tx transaction.NewUndoTx() <- It is made sure that tx is declared once
+// 2. tx.Begin() <- always inserted
+// Other statements to Log data, End the transaction & release the transaction
+// handle will be inserted to application code at a later compilation stage.
+// These statements are inserted by parser so that context, typesystem info for
+// transaction package will be easily populated during AST creation this way
+func (p *parser) txBlockStmt(declTx bool) *TxBlockStmt {
+	t := new(TxBlockStmt)
+	t.pos = p.pos()
+	var sList []Stmt
+	p.next()
+	p.want(_Lparen)
+	p.want(_Rparen) // Not expecting any argument in txn() use
+
+	var s string
+	if declTx {
+		s = "var tx transaction.TX"
```

Comment:
Can we use a more convoluted variable name here (e.g. ___tx)? This will help ensure that we do not use a variable name that the user may already have in his code.